WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/RefundShippingLabelViewController.swift

```diff
@@ -0,0 +1,204 @@
+import UIKit
+import Yosemite
+
+/// Displays information about the refund with a CTA to request a refund.
+final class RefundShippingLabelViewController: UIViewController {
+    @IBOutlet private weak var tableView: UITableView!
+    @IBOutlet private weak var refundButton: UIButton!
+
+    private let shippingLabel: ShippingLabel
+    private let viewModel: RefundShippingLabelViewModel
+    private let rows: [Row]
+    private let noticePresenter: NoticePresenter
+    private let onDismiss: () -> Void
+
+    init(shippingLabel: ShippingLabel,
+         currencyFormatter: CurrencyFormatter = .init(currencySettings: ServiceLocator.currencySettings),
+         noticePresenter: NoticePresenter = ServiceLocator.noticePresenter,
+         onDismiss: @escaping () -> Void) {
+        self.shippingLabel = shippingLabel
+        self.viewModel = RefundShippingLabelViewModel(shippingLabel: shippingLabel, currencyFormatter: currencyFormatter)
+        self.rows = [.purchaseDate, .refundableAmount]
+        self.noticePresenter = noticePresenter
+        self.onDismiss = onDismiss
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureNavigationBar()
+        configureTableView()
+        configureRefundButton()
+    }
+}
+
+// MARK: Action Handling
+private extension RefundShippingLabelViewController {
+    func refundShippingLabel() {
+        let action = ShippingLabelAction.refundShippingLabel(shippingLabel: shippingLabel) { result in
+            self.showRefundResultNotice(result: result)
+        }
+        ServiceLocator.stores.dispatch(action)
+        onDismiss()
+    }
+
+    func showRefundResultNotice(result: Result<Yosemite.ShippingLabelRefund, Error>) {
+        let notice: Notice
+        switch result {
+        case .success:
+            let title = String.localizedStringWithFormat(Localization.refundSuccessNoticeFormat, shippingLabel.serviceName, viewModel.refundableAmount)
+            notice = Notice(title: title, feedbackType: .success)
+        case .failure(let error):
+            DDLogError("⛔️ Failed to request a refund for shipping label \(shippingLabel.shippingLabelID): \(error)")
+            notice = Notice(title: Localization.refundErrorNotice, feedbackType: .error)
+        }
+        noticePresenter.enqueue(notice: notice)
+    }
+}
+
+// MARK: Configuration
+private extension RefundShippingLabelViewController {
+    func configureNavigationBar() {
+        navigationItem.title = Localization.navigationBarTitle
+    }
+
+    func configureTableView() {
+        tableView.dataSource = self
+        tableView.delegate = self
+        registerTableViewCellsAndHeader()
+
+        tableView.tableFooterView = UIView()
```

Comment:
nit: I think we have a helper called `tableView.applyFooterViewForHidingExtraRowPlaceholders` is that what you are trying to achieve?

Reply:
yes thanks for the suggestion! updated in https://github.com/woocommerce/woocommerce-ios/pull/3306/commits/9bd8ddd9b7ba9447f064a10049b4bca1cdde5a17